Address-Book.go

```diff
@@ -41,112 +42,116 @@ type UserContacts struct {
+	var currentcontact Contact
+	var newcontact Contact
+	var phone PhoneNum
 
 	for rows.Next() {
 
-		rows.Scan(&NewContact.Id, &NewContact.FirstName, &NewContact.LastName , &NewContact.Email , &Phone.Id , &Phone.Phonenumber )
+		rows.Scan(&newcontact.Id, &newcontact.FirstName, &newcontact.LastName , &newcontact.Email , &phone.Id , &phone.Phonenumber )
 
-		if NewContact.Id!=CurrentContact.Id && CurrentContact.Id != 0{
+		if newcontact.Id!=currentcontact.Id && currentcontact.Id != 0{
 
-			Context.User.Contacts = append(Context.User.Contacts, CurrentContact)
-			CurrentContact = NewContact
+			user.Contacts = append(user.Contacts, currentcontact)
+			currentcontact = newcontact
 
 
-		}else if CurrentContact.Id == 0{
+		}else if currentcontact.Id == 0{
 
-			CurrentContact=NewContact
+			currentcontact=newcontact
 
 		}
-		CurrentContact.PhoneNumber = append(CurrentContact.PhoneNumber, Phone)
+		currentcontact.PhoneNumber = append(currentcontact.PhoneNumber, phone)
 	}
-	Context.User.Contacts = append(Context.User.Contacts, CurrentContact)
+	user.Contacts = append(user.Contacts, currentcontact)
 	return err
 }
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
-func QueryUser(Context *GlobalData) (string,error){
-	var databaseUsername string
+func QueryUser(context *GlobalData , username string) (string,error){
 	var databasePassword string
 
-	err := Context.db.QueryRow("SELECT username, password FROM users WHERE username=?", Context.User.UserName).Scan(&databaseUsername, &databasePassword)
+	err := context.db.QueryRow("SELECT password FROM users WHERE username=?", username).Scan( &databasePassword)
 
 	return databasePassword,err
 }
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
-func InsertNewContact(Context *GlobalData, w http.ResponseWriter,r *http.Request) (Contact,error){
+func (user * UserContacts) InsertNewContact(context *GlobalData , w http.ResponseWriter,r *http.Request) (Contact,error){
 //Start Transaction
 
-_ , err := Context.db.Exec("START TRANSACTION")
+_ , err := context.db.Exec("START TRANSACTION")
 if err!=nil {
 return Contact{},err
 }
-fmt.Println(Context.User.Id)
-_, err = Context.db.Exec("insert into contact values(? ,? ,? ,? ,? ) ", nil, r.FormValue("first-name"), r.FormValue("last-name"), r.FormValue("email"), Context.User.Id)
+_, err = context.db.Exec("insert into contact values(? ,? ,? ,? ,? ) ", nil, r.FormValue("first-name"), r.FormValue("last-name"), r.FormValue("email"), user.Id)
 if err != nil {
-	Context.db.Exec("ROLLBACK")
+	context.db.Exec("ROLLBACK")
 	return Contact{},err
 }
-
-row := Context.db.QueryRow("select MAX(contactID) from contact")
+// ana msh fahem y3ni eh "you didn't use Auto Increment column"
+// contactID da AutoIncrement fel DB ana lesa 3aml insert fo2 le contact
+//fa bageb ID bta3 a5er contact defto 3shan ast5demo !
+row := context.db.QueryRow("select MAX(contactID) from contact")
```

Comment:
Tayeb kont ool aw es2al :D https://mariadb.com/kb/en/mariadb/auto_increment/. Men8er el auto increment 7aye7ssal mashakel we momken ids tetkarar.